dev/devicelab/bin/tasks/flutter_attach_test_fuchsia.dart

```diff
@@ -0,0 +1,236 @@
+    }
+
+    final Directory fuchsiaToolDirectory =
+      Directory(path.join(flutterDirectory.path, 'bin', 'cache', 'artifacts', 'fuchsia', 'tools'));
+    if (!fuchsiaToolDirectory.existsSync()) {
+      throw Exception('Expected Fuchsia tool directory at ${fuchsiaToolDirectory.path}');
+    }
+
+    final Device device = await devices.workingDevice;
+    final Directory appDir = dir(path.join(
+      flutterDirectory.path,
+      'dev',
+      'integration_tests',
+      'ui',
+    ));
+
+    await inDirectory(appDir, () async {
+      final Random random = Random();
+      final Map<String, Completer<void>> sentinelMessage = <String, Completer<void>>{
+        'sentinel-${random.nextInt(1<<32)}': Completer<void>(),
+        'sentinel-${random.nextInt(1<<32)}': Completer<void>(),
+      };
+
+      Process runProcess;
+      Process logsProcess;
+
+      try {
+        section('Creating lib/fuchsia_main.dart');
+
+        generateMain(appDir, sentinelMessage.keys.toList()[0]);
+
+        section('Launching `flutter run` in ${appDir.path}');
+
+        runProcess = await startProcess(
+          flutterBinary,
+          <String>[
+            'run',
+            '--suppress-analytics',
+            '-d', device.deviceId,
+            '-t', 'lib/fuchsia_main.dart',
+          ],
+          isBot: false, // We just want to test the output, not have any debugging info.
+        );
+
+        logsProcess = await startProcess(
+          flutterBinary,
+          <String>['logs', '--suppress-analytics', '-d', device.deviceId],
+          isBot: false, // We just want to test the output, not have any debugging info.
+        );
+
+        Future<dynamic> eventOrExit(Future<void> event) {
+          return Future.any<dynamic>(<Future<dynamic>>[
+            event,
+            runProcess.exitCode,
+            logsProcess.exitCode,
+          ]);
+        }
+
+        logsProcess.stdout
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String log) {
+            print('logs:stdout: $log');
+            for (final String sentinel in sentinelMessage.keys) {
+              if (log.contains(sentinel)) {
+                if (sentinelMessage[sentinel].isCompleted) {
+                  throw Exception(
+                    'Expected a single `$sentinel` message in the device log, but found more than one'
+                  );
+                }
+                sentinelMessage[sentinel].complete();
+                break;
+              }
+            }
+          });
+
+        final Completer<void> hotReloadCompleter = Completer<void>();
+        final Completer<void> reloadedCompleter = Completer<void>();
+        runProcess.stdout
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+            print('run:stdout: $line');
+            if (line.contains('To detach, press "d"; to quit, press "q"')) {
```

Comment:
This line is different now. See https://github.com/flutter/flutter/commit/23ce192a8f37ae00af12c510fd103b70405e7a3d